src/librustc/hir/intravisit.rs

```diff
@@ -773,9 +773,7 @@ pub fn walk_fn_kind<'v, V: Visitor<'v>>(visitor: &mut V, function_kind: FnKind<'
         FnKind::ItemFn(_, generics, ..) => {
             visitor.visit_generics(generics);
         }
-        FnKind::Method(_, sig, ..) => {
-            visitor.visit_generics(&sig.generics);
-        }
+        FnKind::Method(..) |
```

Comment:
Note that this is a removal. If you trace the code up to the usages of `walk_fn`, I don't see where else this is used. I think it should be alright, but it's worth noting the consequences of this change.

Reply:
I'll try to do a look to get a feel for whether there will be impact -- but as we said before, really this call is just *moving* into `visit_trait_item` and `visit_impl_item`, so it's not like it's a pure removal.